drv/gimlet-hf-server/src/main.rs

```diff
@@ -211,8 +228,10 @@ impl ServerImpl {
 
     fn page_program_raw(&self, addr: u32, data: &[u8]) -> Result<(), HfError> {
         self.set_and_check_write_enable()?;
-        self.qspi.page_program(addr, data);
-        self.poll_for_write_complete(None);
+        self.qspi
+            .page_program(addr, data)
+            .map_err(|x| qspi_to_hf(x))?;
```

Comment:
Nit: you can write this without the lambda, e.g. `map_err(qspi_to_hf)`

(same note applies in a bunch of other places in the PR)

Reply:
I took the suggestion for `read_status`. Maybe it should apply elsehwere?